File: CoWorkHub/CoWorkHub.Model/PagedResult.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class PagedResult<T>
    {
        public IList<T> ResultList { get; set; }
        public int? Count { get; set; }
        public int? Page { get; set; }
        public int? PageSize { get; set; }
        public int? TotalPages { get; set; }
        public bool? HasPreviousPage { get; set; }
        public bool? HasNextPage { get; set; }
    }
}

File: CoWorkHub/UI/coworkhub_mobile/lib/models/paged_result.dart
import 'package:json_annotation/json_annotation.dart';

part 'paged_result.g.dart';

@JsonSerializable(genericArgumentFactories: true)
class PagedResult<T> {
  List<T> resultList;
  int? count;
  int? page;
  int? pageSize;
  int? totalPages;
  bool? hasPreviousPage;
  bool? hasNextPage;

  PagedResult({
    required this.resultList,
    this.count,
    this.page,
    this.pageSize,
    this.totalPages,
    this.hasPreviousPage,
    this.hasNextPage,
  });

  factory PagedResult.fromJson(
    Map<String, dynamic> json,
    T Function(Object? json) fromJsonT,
  ) => _$PagedResultFromJson(json, fromJsonT);

  Map<String, dynamic> toJson(Object Function(T value) toJsonT) =>
      _$PagedResultToJson(this, toJsonT);
}
